### PR TITLE
bpo-39883: Use BSD0 license for code in docs

### PR DESCRIPTION
**Pull request by toddrjen**
**NOTE** Please do *not* accept this pull request.  It needs to be discussed and approved by the PSF board. (**UPDATE:** The PSF board approved it. We are going ahead.)

On the Python-ideas mailing list, we [discussed](https://mail.python.org/archives/list/python-ideas@python.org/thread/OZDSGC3I72J4GU375L7FLBLLTYXNLE34/#GMIRRC56RPHDSNQ75KNZQ466U3BE7KZY) making the code in the documentation under a more permissive license to make it easier for people to re-use it.   Specifically, the idea is to use the [BSD zero clause](https://spdx.org/licenses/0BSD.html) license to remove essentially all restrictions on the use of that code.

Guido van Rossum liked the idea, so per his request I am submitting a pull request as the first stage in getting this change potentially implemented.  I will then submit this pull request, and the accompanying pull request python/python-docs-theme#36, to the PSF board for discussion.  It will also likely need a review by a lawyer.

This specific request adds additional information on what, specifically, the current license is, the part about code in the documentation being under the 0BSD license, the text of the 0BSD license, and a bit about other parts being under other licenses (I am particularly unsure that the last part should be in there at all).

The current format will likely change substantially, and changes to the CPython side may not end up being necessary.  Guido seemed to think that simply a blurb in the online documentation footer may be sufficient.  The code that generates the footer is in the [python/python-docs-theme](https://github.com/python/python-docs-theme/) project and the change has been submitted there in pull request python/python-docs-theme#36.

Comments are appreciated. 

*Edit*: Added information about additional pull request.

https://bugs.python.org/issue39883

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

## Recognized GitHub username

We couldn't find a [bugs.python.org](https://bugs.python.org/) (b.p.o) account corresponding to the following GitHub usernames:

@toddrjen

This might be simply due to a missing "GitHub Name" entry in one's b.p.o account settings. This is necessary for legal reasons before we can look at this contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.




You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

File: Doc/license.rst
Comment (gvanrossum) [inline Doc/license.rst]: I presume X.X.X will be 3.9.0. (Also in the LICENSE file.)

Comment (toddrjen) [reply Doc/license.rst]: I wasn't sure how long this process will take so I didn't want to commit to a particular release.  if it makes it into 3.9.0 that would be great.

**Comment by Mariatta** [comment]
What does PSF legal team say about this? Cc @VanL

**Comment by VanL** [comment]
This is fine, but our contributor agreement says that it needs to be by vote of the board. The vote will happen later this month, I don't expect it will be an issue.

Thanks,
Van

**Comment by Mariatta** [comment]
Thanks for the reply @VanL for the quick reply.

I think this is change will be worthy of a News Entry.

**Comment by gvanrossum** [comment]
@toddrjen Do you know how to add a news entry? (https://discuss.python.org/t/blurb-it-is-now-available/528)

**Comment by toddrjen** [comment]
@gvanrossum I made a blurb.  Please let me know if it is okay.

**Comment by gvanrossum** [comment]
Yup, thanks! Now we wait for the board...

File: Doc/license.rst
Comment (terryjreedy) [reply Doc/license.rst]: I would think that this could be backported.  If it is, they should also be listed.  In any case, good idea.

**Comment by toddrjen** [comment]
Just following up on this.  Did the board have a vote?  I don't see one mentioned in the minutes.

**Comment by gvanrossum** [comment]
I’ve pinged Ewa.

**Comment by ejodlowska** [comment]
Hi folks - the following resolution was approved by the PSF Board on August 26, 2020: RESOLVED: The license for code included within the Python documentation is changed to Zero-Clause BSD / Free Public License 1.0.0 (0BSD), effective as of the date of this resolution.

**Comment by gvanrossum** [comment]
Thanks Ewa!

Todd, were you interested in submitting PRs to implement this change?

File: Doc/license.rst
Comment (gvanrossum) [reply Doc/license.rst]: I think we can list this as per 3.9.0 even if it doesn't make it in the release (rc2 is scheduled for Sept 14).

Release manager (@ambv) do you think it's reasonable to land this doc-only change?

And I agree we could backport to 3.8.0, in which case we should just say per 3.8.6 (which would be the next release, also going out Sept 14).

File: LICENSE
Comment (gvanrossum) [inline LICENSE]: Another XXX to fix.

**Review by gvanrossum — COMMENTED** [review]
Sorry, I forgot that this *is* the PR. I will approve once you put the version numbers in. I think 3.8.6 is reasonable.

File: Doc/license.rst
Comment (ambv) [reply Doc/license.rst]: @gvanrossum Perfectly reasonable.

Comment (ambv) [reply Doc/license.rst]: (Both for this to land in 3.8.6rc1 as well as in 3.9.0rc2.)

File: LICENSE
Comment (toddrjen) [reply LICENSE]: Fixed

**Comment by toddrjen** [comment]
@gvanrossum I set the two version numbers to 3.8.6 as requested.

File: LICENSE
Comment (gvanrossum) [inline LICENSE]: Drop ‘|release|’.

File: LICENSE
Comment (gvanrossum) [inline LICENSE]: Drop this line.

File: Doc/license.rst
Comment (gvanrossum) [inline Doc/license.rst:78]: Maybe insert blank line?

**Review by gvanrossum — COMMENTED** [review]
Thanks! Almost there; I found some markup on LICENSE.

File: LICENSE
Comment (toddrjen) [reply LICENSE]: Fixed

File: LICENSE
Comment (toddrjen) [reply LICENSE]: Fixed

File: Doc/license.rst
Comment (toddrjen) [reply Doc/license.rst:78]: Fixed

**Review by gvanrossum — APPROVED** [review]

**Pull request merged**
